### PR TITLE
lib: lte_link_control: add %PALL

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
Adds support for the `PALL` AT command.

- [x] Doxygen for struct, type, macros
- [ ] Changelog
- [ ] Update library documentation
- [ ] Check code coverage
- [ ] Split into library commit and modem shell commit

### CODE REVIEW

File: lib/lte_link_control/modules/pall.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	char *plmn_list_str_format;
+	char entry_str_temp[LTE_LC_PLMN_ENTRY_STR_SIZE + 1] = { 0 };
+
+	if (!plmn_list) {
+		LOG_ERR("Invalid arguments.");
+		return -EINVAL;
+	}
+
+	if (size > LTE_LC_PLMN_LIST_MAX) {
+		LOG_ERR("Invalid PLMN access list size.");
```
Comment (eivindj-nordic) [inline lib/lte_link_control/modules/pall.c:35]:
```suggestion
		LOG_ERR("PLMN access list size too big.");
```

File: lib/lte_link_control/modules/pall.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	struct at_parser parser = { 0 };
+	char *response_buffer;
+
+	if (!plmn_list || !size || !list_type) {
+		LOG_ERR("Invalid arguments.");
+		return -EINVAL;
+	}
+
+	if (*size > LTE_LC_PLMN_LIST_MAX) {
+		LOG_ERR("Invalid PLMN access list size.");
```
Comment (eivindj-nordic) [inline lib/lte_link_control/modules/pall.c:84]:
```suggestion
		LOG_ERR("PLMN access list size too big.");
```

File: samples/cellular/modem_shell/src/link/link_shell.c [hunk 3]
```diff
@@ -369,6 +370,31 @@ static const char link_modem_usage_str[] =
 	"\n"
 	"Several options can be given and they are run in the given order.";
 
+static const char link_pall_usage_str[] =
+	"Usage: link pall --read | --write [options] | --clear\n"
+	"Options:\n"
+	"  -r, --read,            Read a PLMN access list.\n"
+	"      --write,           Write a PLMN access list.\n"
```
Comment (eivindj-nordic) [inline samples/cellular/modem_shell/src/link/link_shell.c:377]: Should there be a short for write?
```suggestion
	"  -w, --write,           Write a PLMN access list.\n"
```

Comment (MirkoCovizzi) [reply samples/cellular/modem_shell/src/link/link_shell.c:377]: I don't think it's supported. I copied read and write from other commands.

Comment (eivindj-nordic) [reply samples/cellular/modem_shell/src/link/link_shell.c:377]: ok

File: samples/cellular/modem_shell/src/link/link_shell.c [hunk 4]
```diff
@@ -50,7 +50,8 @@ enum link_shell_command {
 	LINK_CMD_DNSADDR,
 	LINK_CMD_REDMOB,
 	LINK_CMD_PROPRIPSM,
-	LINK_CMD_MODEM
+	LINK_CMD_MODEM,
+	LINK_CMD_PALL,
```
Comment (tokangas) [inline samples/cellular/modem_shell/src/link/link_shell.c:54]: The implementation has of course already done, but I'm just wondering what is the benefit of having this functionality in modem_shell. The command format when writing the list using the new command is not that different from the AT command, at least the PLMN/access technology list part.

File: samples/cellular/modem_shell/src/link/link_shell.c [hunk 5]
```diff
@@ -369,6 +370,32 @@ static const char link_modem_usage_str[] =
...
+	"                         Either the --allowed option or the --unallowed option are"
+	"                         required.\n"
+	"                         Use --plmn_list option to specify the PLMN access list\n"
+	"                         to configure. The --plmn_list option is required.\n"
+	"      --allowed          Option for configuring the list of allowed PLMNs.\n"
+	"                         This option is required in order to use --write.\n"
+	"      --unallowed        Option for configuring the list of unallowed PLMNs.\n"
+	"                         This option is required in order to use --write.\n"
+	"      --plmn_list, [str] PLMN access list to configure. Usage:\n"
+	"                         <plmn_1>,<act_1>[,<plmn_2>,<act_2]...[,<plmn_25>,<act_25>]\n"
```
Comment (tokangas) [inline samples/cellular/modem_shell/src/link/link_shell.c:393]: `act` parameter format should be described.

File: samples/cellular/modem_shell/src/link/link_shell.c (hunk 4 above)
Comment (MirkoCovizzi) [reply samples/cellular/modem_shell/src/link/link_shell.c:54]: Yeah, we just had a brief discussion about this on Teams and it seems that there might not even be an actual need of this functionality at such a high level API...

**Comment by MirkoCovizzi** [comment]
Reverting back to draft because of https://github.com/nrfconnect/sdk-nrf/pull/17359#discussion_r1820778011

Might be updated in the future if there is a need for it.

**Pull request closed without merging**